app/controllers/discovered_hosts_controller.rb

```diff
@@ -20,6 +20,15 @@ def model_of_controller
     Host::Discovered
   end
 
+  def welcome
+    @hosts = Host::Discovered.all
+    if @hosts.empty?
+      redirect_to :controller => 'foreman_discovery/react', :action => 'index'
```

Comment:
@lzap @ezr-ondrej I am trying to redirect to the `foreman_discovery/react` controller I have written for react but it doesn't seem to redirect. Can you take a look please?

Reply:
There is no route for react controller action in the config/routes, it will redirect if you add something like:

```rb
get 'discovered_hosts/empty', to: 'foreman_discovery/react#index'
```

That will redirect you to a completely new page (with different URL) that shows the empty state:

![](https://i.imgur.com/fHzdUoB.png)

I think you want to stay on the same page, just displaying the empty component. You can do that by mounting the EmptyState component directly into the welcome.html.erb

Reply:
Awesome! thanks @xprazak2  :) It works like a charm!